src/odemis/gui/util/img.py

```diff
@@ -112,12 +108,60 @@ def format_rgba_darray(im_darray, alpha=None):
         rgba[:, :, 1] = im_darray[:, :, 1]
         rgba[:, :, 2] = im_darray[:, :, 0]
         rgba[:, :, 3] = im_darray[:, :, 3]
-        new_darray = model.DataArray(rgba)
-        new_darray.metadata['byteswapped'] = True
-        return new_darray
     else:
         raise ValueError("Unsupported colour depth!")
 
+    new_darray = model.DataArray(rgba)
+    new_darray.metadata['byteswapped'] = True
+    return new_darray
+
+
+def format_bgra_to_rgb(im_darray, keepalpha=True, inplace=False):
+    """ Reshape the given numpy.ndarray from RGB to BGRA format
+    im_darray (DataArray of shape Y,X,{3,4}): input image
+    keepalpha (bool): If an alpha is present, keep it. IOW, if there are 4 channels,
+      the 4th channel will be kept as is.
+    inplace (bool): directly modify im_darray
+    return (DataArray of shape Y,X,{3,4}): The return type is the same of im_darray
+    """
+    assert im_darray.ndim == 3
+
+    if hasattr(im_darray, 'metadata'):
+        if not im_darray.metadata.get('byteswapped', True):
+            logging.warning("Trying to convert to RGB an array already in RGB")
+            return im_darray
+
+    if im_darray.shape[-1] == 3:
+        shape = im_darray.shape
+    elif im_darray.shape[-1] == 4:
+        if keepalpha:
+            shape = im_darray.shape
+        else:
+            shape = im_darray.shape[:2] + (3,)
+            if inplace:
+                raise ValueError("Cannot drop alpha channel in-place")
+    else:
+        raise ValueError("Unsupported colour depth!")
+
+    if inplace:
+        rgba = im_darray
+        rgba[:,:, [2, 0]] = rgba[:,:, [0, 2]]  # Just switch R <> B
```

Comment:
Just a thought on this parameter/part of the code:
If the image is modified "in_place" and you don't use the output value (as in a part of your current code), do you still want to return the array?

Reply:
Yes, I did this on purpose. I agree this is arguable, but my idea is that I want the function to always have the same signature. It avoids surprises. That's actually what numpy functions do as well (when you pass a `out` parameter).